### PR TITLE
refactor: refactor checkout steps and order confirmation and order details page

### PR DESCRIPTION
**Pull request by WeizhengSap**
After pickup feature is enabled, the pickup data or items will be displayed in checkout steps, order confirmation and order details pages.

### CODE REVIEW

File: projects/storefrontapp/src/environments/environment.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { Environment } from './models/environment.model';
 
 export const environment: Environment = {
   production: false,
-  occBaseUrl: buildProcess.env.CX_BASE_URL,
+  occBaseUrl: 'https://spartacus-dev0.eastus.cloudapp.azure.com:9002',
```
Comment (WeizhengSap) [inline projects/storefrontapp/src/environments/environment.ts]: don't forget to revert it back before merging.

File: feature-libs/checkout/b2b/components/checkout-review-submit/checkout-review-submit.component.html [hunk 2]
```diff
@@ -184,10 +184,7 @@ <h2 class="cx-review-title d-none d-lg-block d-xl-block">
     <div class="cx-review-cart-heading d-block d-lg-none d-xl-none">
       {{ 'checkoutReview.placeOrder' | cxTranslate }}
     </div>
-    <div
-      class="cx-review-cart-item col-md-12"
-      *ngIf="entries$ | async as entries"
-    >
+    <div class="cx-review-cart-item" *ngIf="entries$ | async as entries">
```
Comment (plabadie) [inline feature-libs/checkout/b2b/components/checkout-review-submit/checkout-review-submit.component.html:187]: Likely a change that should show in the migration docs.  Adding classes is in the list of breaking changes.

File: feature-libs/checkout/b2b/components/checkout-delivery-address/checkout-delivery-address.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <h2 class="cx-checkout-title d-none d-lg-block d-xl-block">
-  {{ 'checkoutAddress.deliveryAddress' | cxTranslate }}
+  {{ 'checkoutAddress.shippingAddress' | cxTranslate }}
```
Comment (plabadie) [inline feature-libs/checkout/b2b/components/checkout-delivery-address/checkout-delivery-address.component.html:2]: Could this be a breaking change?  If so, needs needs doc.

File: feature-libs/checkout/base/assets/translations/en/checkout.i18n.ts [hunk 4]
```diff
@@ -10,13 +10,13 @@ export const checkout = {
   },
   checkoutProgress: {
     label: 'Checkout Progress',
-    deliveryAddress: 'Delivery Address',
```
Comment (plabadie) [inline feature-libs/checkout/base/assets/translations/en/checkout.i18n.ts:13]: Changing labels are breaking changes.  Need entries in migration doc.

File: feature-libs/checkout/b2b/components/guards/checkout-b2b-steps-set.guard.ts [hunk 5]
```diff
@@ -44,6 +45,7 @@ export class CheckoutB2BStepsSetGuard
     protected checkoutPaymentFacade: CheckoutPaymentFacade,
     protected checkoutDeliveryModesFacade: CheckoutDeliveryModesFacade,
     protected router: Router,
+    protected activeCartFacade: ActiveCartFacade,
```
Comment (plabadie) [inline feature-libs/checkout/b2b/components/guards/checkout-b2b-steps-set.guard.ts]: Why add it in the middle of the param list?  It makes it harder to migrate.

File: feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.html [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <h2 class="cx-checkout-title d-none d-lg-block d-xl-block">
-  {{ 'checkoutAddress.deliveryAddress' | cxTranslate }}
+  {{ 'checkoutAddress.shippingAddress' | cxTranslate }}
```
Comment (plabadie) [inline feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.html:2]: Same as other comment.  Potential breaking change.  Needs migration doc.

File: feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.html [hunk 7]
```diff
@@ -4,63 +4,78 @@ <h2 class="cx-checkout-title d-none d-lg-block d-xl-block">
 
 <ng-container
   *ngIf="
-    (supportedDeliveryModes$ | async)?.length && !(isUpdating$ | async);
+    !(isUpdating$ | async) &&
```
Comment (plabadie) [inline feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.html:7]: Dom changes, need migration docs.

Please review the whole PR for dom changes and add corresponding migration doc.

File: feature-libs/checkout/base/styles/components/order-confirmation/_order-confirmation-thank-you-message.scss [hunk 8]
```diff
@@ -4,7 +4,8 @@
     padding: var(--cx-padding, 2.5rem);
 
     h2 {
-      font-weight: $font-weight-normal;
+      font-weight: var(--cx-font-weight-bold);
+      font-size: var(--cx-font-size, 1.5rem);
```
Comment (plabadie) [inline feature-libs/checkout/base/styles/components/order-confirmation/_order-confirmation-thank-you-message.scss:8]: Changes to existing CSS styles.  Please review the PR for other instances of changes to styles and add migration docs for each.

File: feature-libs/order/components/order-details/order-details.module.ts [hunk 9]
```diff
@@ -70,21 +71,45 @@ const moduleComponents = [
...
+          component: OrderOverviewComponent,
+          guards: [AuthGuard],
+          data: {
+            simple: true,
+          },
         },
+        // AccountOrderDetailsBillingComponent: {
+        //   component: OrderDetailBillingComponent,
+        //   guards: [AuthGuard],
+        // },
```
Comment (plabadie) [inline feature-libs/order/components/order-details/order-details.module.ts]: Is this dead code or something that should have been uncommented?  Either remove the dead code or uncomment it.

File: feature-libs/order/root/facade/order.facade.ts [hunk 10]
```diff
@@ -41,4 +44,13 @@ export abstract class OrderFacade {
...
+  /**
+   * Return cart's pickup entries
+   */
+  abstract getPickupEntries(): Observable<OrderEntry[]>;
+
+  /**
+   * Return cart's delivery entries
+   */
+  abstract getDeliveryEntries(): Observable<OrderEntry[]>;
 }
```
Comment (plabadie) [inline feature-libs/order/root/facade/order.facade.ts]: The doc comments mention the cart, should it be the order instead?

File: feature-libs/order/styles/components/order-details/_order-details-shipping.scss [hunk 11]
```diff
@@ -1,4 +1,4 @@
-%cx-order-details-shipping {
```
Comment (plabadie) [inline feature-libs/order/styles/components/order-details/_order-details-shipping.scss:1]: Probably a breaking change.

File: feature-libs/organization/order-approval/components/details/order-approval-details.module.ts [hunk 12]
```diff
@@ -62,7 +62,7 @@ import { OrderDetailPermissionResultsComponent } from './order-detail-permission
           component: OrderDetailPermissionResultsComponent,
         },
         OrderApprovalDetailShippingComponent: {
-          component: OrderDetailShippingComponent,
```
Comment (plabadie) [inline feature-libs/organization/order-approval/components/details/order-approval-details.module.ts:65]: Should be documented in the ts manual breaking change doc for hicher level changes like this.

File: projects/storefrontapp-e2e-cypress/cypress/e2e/regression/checkout/billing-address.core-e2e-flaky.cy.ts [hunk 13]
```diff
@@ -41,7 +41,7 @@ context('Payment billing address', { testIsolation: false }, () => {
     checkout.fillAddressFormWithCheapProduct();
     checkout.verifyDeliveryMethod();
     checkout.fillPaymentFormWithCheapProduct();
-    checkout.verifyReviewOrderPage();
+    //checkout.verifyReviewOrderPage();
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/regression/checkout/billing-address.core-e2e-flaky.cy.ts]: The dead code in this file need to be handled.

File: feature-libs/order/root/facade/order.facade.ts (hunk 10 above)
Comment (WeizhengSap) [reply feature-libs/order/root/facade/order.facade.ts]: yes

File: projects/storefrontapp/src/environments/environment.ts [hunk 14]
```diff
@@ -21,7 +21,7 @@ import { Environment } from './models/environment.model';
...
   occApiPrefix: '/occ/v2/',
```
Comment (plabadie) [inline projects/storefrontapp/src/environments/environment.ts:25]: Just checking, was temp config thing reverted?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/express-checkout.ts [hunk 15]
```diff
@@ -23,21 +23,21 @@ export function testExpressCheckout() {
...
-    cy.get('.cx-checkout-title').should('contain', 'Delivery Address');
+    cy.get('.cx-checkout-title').should('contain', 'Shipping Address');
   });
 
   it('should skip address and payment checkout steps once address and payment are set', () => {
     checkout.fillAddressFormWithCheapProduct();
     checkout.verifyDeliveryMethod();
     checkout.fillPaymentFormWithCheapProduct();
-    checkout.verifyReviewOrderPage();
+    //checkout.verifyReviewOrderPage();
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/express-checkout.ts]: The dead code in this file needs to be handled.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts [hunk 16]
```diff
@@ -85,11 +85,9 @@ export function fillShippingAddress(
           .type(shippingAddress.address.postal);
       shippingAddress?.cellphone &&
         cy
-          .get('[formcontrolname="cellphone"]')
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts:88]: No more cell phone?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts [hunk 17]
```diff
@@ -257,12 +257,16 @@ export function fillPaymentForm(
...
+}
+
 export function verifyReviewOrderPage() {
-  cy.get('.cx-review-title').should('contain', 'Review');
+  cy.contains('Cart total');
 }
 
 export function placeOrder() {
-  verifyReviewOrderPage();
+  //verifyReviewOrderPage();
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts]: Dead code that need handling.

**Review by plabadie — CHANGES_REQUESTED** [review]
See comments on the PR

File: projects/storefrontapp/src/environments/environment.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontapp/src/environments/environment.ts]: Good to have this reminder.  It doesn't look reverted.

File: feature-libs/checkout/b2b/components/checkout-review-submit/checkout-review-submit.component.html (hunk 2 above)
Comment (RadhepS) [reply feature-libs/checkout/b2b/components/checkout-review-submit/checkout-review-submit.component.html:187]: Added

File: feature-libs/checkout/b2b/components/checkout-delivery-address/checkout-delivery-address.component.html (hunk 3 above)
Comment (RadhepS) [reply feature-libs/checkout/b2b/components/checkout-delivery-address/checkout-delivery-address.component.html:2]: Added

File: feature-libs/checkout/base/assets/translations/en/checkout.i18n.ts (hunk 4 above)
Comment (RadhepS) [reply feature-libs/checkout/base/assets/translations/en/checkout.i18n.ts:13]: Added

File: feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.html (hunk 6 above)
Comment (RadhepS) [reply feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.html:2]: Added

File: feature-libs/checkout/base/styles/components/order-confirmation/_order-confirmation-thank-you-message.scss (hunk 8 above)
Comment (RadhepS) [reply feature-libs/checkout/base/styles/components/order-confirmation/_order-confirmation-thank-you-message.scss:8]: Added

File: feature-libs/organization/order-approval/components/details/order-approval-details.module.ts (hunk 12 above)
Comment (RadhepS) [reply feature-libs/organization/order-approval/components/details/order-approval-details.module.ts:65]: Added

File: projects/storefrontapp-e2e-cypress/cypress/helpers/express-checkout.ts (hunk 15 above)
Comment (RadhepS) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/express-checkout.ts]: Fixed

File: projects/storefrontapp-e2e-cypress/cypress/e2e/regression/checkout/billing-address.core-e2e-flaky.cy.ts (hunk 13 above)
Comment (RadhepS) [reply projects/storefrontapp-e2e-cypress/cypress/e2e/regression/checkout/billing-address.core-e2e-flaky.cy.ts]: Fixed

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts (hunk 17 above)
Comment (RadhepS) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts]: Fixed

File: feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.html (hunk 7 above)
Comment (RadhepS) [reply feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.html:7]: Added

File: feature-libs/order/components/order-details/order-details.module.ts (hunk 9 above)
Comment (WeizhengSap) [reply feature-libs/order/components/order-details/order-details.module.ts:109]: We have cms component `OrderDetailBillingComponent`, but in the UI design, the billing is part of overview. I will remove it.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts (hunk 16 above)
Comment (WeizhengSap) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-forms.ts:88]: just phone.

File: projects/storefrontapp/src/environments/environment.ts (hunk 1 above)
Comment (WeizhengSap) [reply projects/storefrontapp/src/environments/environment.ts]: reverted

File: feature-libs/order/components/order-details/order-details.module.ts (hunk 9 above)
Comment (WeizhengSap) [reply feature-libs/order/components/order-details/order-details.module.ts]: removed

**Review by plabadie — DISMISSED** [review]

**Review by RadhepS — APPROVED** [review]

**Pull request merged**